front: fix Graou import

emersion opened the pull request:
_See individual commits._

emersion committed:
front: fix UIC type for Graou API

Graou happens to return a string.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: fix secondary code for Graou import

Empty secondary codes result in broken steps.

Signed-off-by: Simon Ser <contact@emersion.fr>

flomonster reviewed (approved):
LGTM, thanks for this fix.

@celine-sncf might know how the control key works 🤷

emersion committed:
front: populate last CI digit for Graou import

Graou now omits the UIC's last digit in train schedule steps for
French stations. The last digit is a checksum computed using the
Luhn algorithm:
https://en.wikipedia.org/wiki/Luhn_algorithm

Implement the algorithm to fix UIC path item locations in imported
train schedules.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: drop unnecessary trigram fallback in Graou import

If a step doesn't have a valid UIC, the Graou import logic falls
back to the trigram.

This fallback was introduced in 02ea4601ad80 ("front: fix open
data import") to handle XML files. Now XML import has been decoupled
from Graou import, but this remnant hasn't been dropped.

Remove the fallback because Graou always returns the UIC.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion commented:
I've now completely fixed Graou imports!

younesschrifi reviewed (commented):
LGTM ! 
thank you for the PR, very easy to review 😄 
I will test it later :)

flomonster reviewed (approved):
Thanks for this PR 🙏

SharglutDev reviewed (approved):
Thanks for the PR 🙇
